AdyenCardScanner/Sources/CardScanner.swift

```diff
@@ -0,0 +1,36 @@
+//
+// Copyright (c) 2025 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+import Foundation
+import UIKit
+
+protocol CardScanning {
+    static func createCardScanner(
+        completion: @escaping (Result<CreditCard, CardScannerError>) -> Void
+    ) -> UIViewController?
+
+    static var isAvailable: Bool { get }
+}
+
+public enum CardScanner: CardScanning {
+
+    // MARK: - Properties
+
+    private static let cardScannerAssembler = CardScannerAssembler()
+
+    // MARK: - CardScanning
+
+    public static func createCardScanner(
+        completion: @escaping (Result<CreditCard, CardScannerError>) -> Void
+    ) -> UIViewController? {
+        let cardScannerViewController = cardScannerAssembler.resolveCardScannerViewController(completion: completion)
```

Comment:
I see `CardScannerAssembling `s function and  `CardScanning ` `createCardScanner(..` have the same signatures. (completion and return values). Do we need 2 different ones then? 
The second layer seems redundant as it's practically the same thing.
We can do the resolving elements inside this block maybe?

Reply:
We don't necessarily need the two layers. Everything can be done directly in the CardScanning one. I just prefer to have anything internal within the assembler layer and use this one is the only public class of the SDK.